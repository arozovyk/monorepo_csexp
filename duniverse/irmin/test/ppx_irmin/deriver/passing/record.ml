(* Records *)
type test_record1 = { alpha : string; beta : int64 list; gamma : unit }
[@@deriving irmin]

type test_record2 = {
  the_FIRST_identifier : test_record1 option;
  the_SECOND_identifier : (string, int32) result list;
}
[@@deriving irmin]
