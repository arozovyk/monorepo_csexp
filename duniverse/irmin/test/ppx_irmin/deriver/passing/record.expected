type test_record1 = {
  alpha: string ;
  beta: int64 list ;
  gamma: unit }[@@deriving irmin]
include
  struct
    let test_record1_t =
      let open Irmin.Type in
        ((((record "test_record1"
              (fun alpha -> fun beta -> fun gamma -> { alpha; beta; gamma }))
             |+ (field "alpha" string (fun t -> t.alpha)))
            |+ (field "beta" (list int64) (fun t -> t.beta)))
           |+ (field "gamma" unit (fun t -> t.gamma)))
          |> sealr
  end[@@ocaml.doc "@inline"][@@merlin.hide ]
type test_record2 =
  {
  the_FIRST_identifier: test_record1 option ;
  the_SECOND_identifier: (string, int32) result list }[@@deriving irmin]
include
  struct
    let test_record2_t =
      let open Irmin.Type in
        (((record "test_record2"
             (fun the_FIRST_identifier ->
                fun the_SECOND_identifier ->
                  { the_FIRST_identifier; the_SECOND_identifier }))
            |+
            (field "the_FIRST_identifier" (option test_record1_t)
               (fun t -> t.the_FIRST_identifier)))
           |+
           (field "the_SECOND_identifier" (list (result string int32))
              (fun t -> t.the_SECOND_identifier)))
          |> sealr
  end[@@ocaml.doc "@inline"][@@merlin.hide ]
